app/page/info.tsx

```diff
@@ -5,120 +5,203 @@ import { Link } from "wouter"
+}
+
 export function Info() {
+  const explorerUrl = "https://jiffyscan.xyz/"
+
   const { provider } = useProviderContext()
+  const sentUserOpStmts = useSentUserOperationStatements()
+  const userOpStmts = useUserOperationStatements()
   const account = useAccount()
+  const network = useNetwork()
+
+  const [node, setNode] = useState<NodeProvider>(
+    new NodeProvider(network.nodeRpcUrl)
+  )
+  const [bundler, setBundler] = useState<BundlerProvider>(
+    new BundlerProvider(
+      network.bundlerRpcUrl,
+      network.chainId === 1337 ? BundlerMode.Manual : BundlerMode.Auto // 1337 is local testnet
+    )
+  )
+
+  const [chainName, setChainName] = useState<string>("")
   const [balance, setBalance] = useState<bigint>(0n)
-  const [transactionHashes, setTransactionHashes] = useState<HexString[]>([""])
-  const [internalTransactionHashes, setInternalTransactionHashes] = useState<
-    HexString[]
-  >([""])
-  const [explorerUrl, setExplorerUrl] = useState<string>("")
+  const [balanceLoading, setBalanceLoading] = useState<boolean>(false)
+  const [userOpsData, setUserOpsData] = useState<UserOperationData[]>([])
+
+  const markUserOperationSucceeded = useStorage(
+    (storage) => storage.markUserOperationSucceeded
+  )
+  const markUserOperationFailed = useStorage(
+    (storage) => storage.markUserOperationFailed
+  )
 
   useEffect(() => {
+    // Retrieve chainName when the Info page is opened
     const asyncFn = async () => {
-      const explorer = getExplorerUrl((await provider.getNetwork()).name)
-      setExplorerUrl(explorer)
+      const chain = getChainName((await provider.getNetwork()).name)
+      setChainName(chain)
+    }
 
+    asyncFn()
+  }, [])
+
+  useEffect(() => {
+    setBalanceLoading(true)
+    // Periodically check the balance of the account
+    const asyncIntervalFn = async () => {
       const balance = await provider.getBalance(account.address)
+      setBalanceLoading(false)
       setBalance(balance)
+    }
+    const id = setInterval(() => {
+      asyncIntervalFn().catch((e) => console.log(e))
+    }, 3333) // Every 3.333 seconds
 
-      const txHashes = await accountTransactions(explorer, account.address)
-      setTransactionHashes(txHashes)
+    return () => {
+      clearInterval(id)
+    }
+  }, [])
+
+  useEffect(() => {
+    // Update userOpsData when userOpStmts change
+    const userOpsData = getUserOpsData(userOpStmts, account.chainId)
+    setUserOpsData(userOpsData)
+  }, [userOpStmts])
 
-      const internalTxHashes = await accountInternalTransactions(
-        explorer,
-        account.address
+  useEffect(() => {
+    // Update node and bundler when network change
+    setNode(new NodeProvider(network.nodeRpcUrl))
+    setBundler(
+      new BundlerProvider(
+        network.bundlerRpcUrl,
+        network.chainId === 1337 ? BundlerMode.Manual : BundlerMode.Auto // 1337 is local testnet
       )
-      setInternalTransactionHashes(internalTxHashes)
+    )
+  }, [network])
```

Comment:
In previous commits [https://github.com/pilagod/waallet-extension/pull/83/commits/5029e2d4ef6b13008ffb67324fbc28fe9cc03783](https://github.com/pilagod/waallet-extension/pull/83/commits/5029e2d4ef6b13008ffb67324fbc28fe9cc03783) of this PR, I tried adding RPC methods, but this would make the provider deviate from using commonly known methods?